### ISSUE TITLE
(mysql backend) - kubectl delete a NotReady node and the node will always come back

### ISSUE THREAD

**Issue by pierredavidbelanger** [rancher/k3s#945]
**Version:**

```
$ /usr/local/bin/k3s --version
k3s version v0.10.0 (f9888ca3)
```

Server:
```
$ curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION='v0.10.0' sh -s - server --data-dir /mnt/k3s --disable-agent --http-listen-port 6080 --storage-endpoint 'mysql://'
```

Agent:
```
$ curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION='v0.10.0' sh -s - agent --server 'https://server.k3s:6443'  --token "$token" --docker --node-name "$instance_id" --node-label "k3s.domain.com/nodepool=${asg_name}"
```

**Describe the bug**

I terminated an agent instance (lets say `i-02bc0484d75f79ed3`) and started a new one (say `i-0a4db90b5e5c461ae`).

As expected, the old node stick around, so I deleted it:
```
$ kubectl get nodes
NAME                  STATUS     ROLES    AGE   VERSION
i-02bc0484d75f79ed3   NotReady   <none>   52m   v1.16.2-k3s.1
i-0a4db90b5e5c461ae   Ready      <none>   14m   v1.16.2-k3s.1

$ kubectl delete node/i-02bc0484d75f79ed3
node "i-02bc0484d75f79ed3" deleted

$ kubectl get nodes
NAME                  STATUS   ROLES    AGE   VERSION
i-0a4db90b5e5c461ae   Ready    <none>   15m   v1.16.2-k3s.1
```

But a couple of seconds (30s maybe) later:
```
$ kubectl get nodes
NAME                  STATUS     ROLES    AGE   VERSION
i-02bc0484d75f79ed3   NotReady   <none>   54m   v1.16.2-k3s.1
i-0a4db90b5e5c461ae   Ready      <none>   16m   v1.16.2-k3s.1
```

Zombie `i-02bc0484d75f79ed3` is always comming back, I can reproduce this again and again.

**To Reproduce**

Terminate an agent node, then `kubectl delete` it.

**Expected behavior**

A deleted node should not come back.

**Actual behavior**

A deleted node will come back again and again.

**Additional context**

I am using a mysql storage.

I confirm that the AWS instances are indeed in the terminated state.

**Comment by erikwilson** [comment]
Sorry about that, Postgres may be a more stable backend to use.

**Comment by davidnuzik** [comment]
Related: https://github.com/rancher/k3s/issues/948

**Comment by ShylajaDevadiga** [comment]
Resolved in v0.11.0-alpha3. Closing issue.
